### PR TITLE
HeadlessUI to Radix

### LINKED ISSUE #2073: replace headlessui with radix-ui
we still have some places where we use headless-ui
image

instead of `radix-ui`. Lets replace those components and remove the dependency.

`import { Switch } from "@headlessui/react";` => https://www.radix-ui.com/docs/primitives/components/switch

`import { Menu, Transition } from "@headlessui/react";` => https://www.radix-ui.com/docs/primitives/components/dropdown-menu

We need a new `<Menu/>` component (similar to `components/ui/Dropdown/Dropdown.tsx`) 

remember to abide by the tailwind design system and not use the default radix-ui styles

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Fixes #2073 

## Type of change

- [X] Technical debt

## How should this be tested?

 - Go to Event type page in your mobile phone and see the mobile-specific dropdown to be Radix and have UI/UX consistency to the rest of menus 👇
Event_Types___Cal_com

- Go to Bookings page in your mobile phone and see the mobile-specific dropdown for a booking to be Radix and have UI/UX consistency to the rest of menus 👇
Bookings___Cal_com

### CODE REVIEW

**Comment by zomars** [comment]
Let's assign reviewers after this PR is ready and a self review is done. 🙏🏽

**Comment by PeerRich** [comment]
> Let's assign reviewers after this PR is ready and a self review is done. 🙏🏽

yeah self review would be great @leog

File: apps/web/components/Dialog.tsx [hunk 1]
```diff
@@ -59,12 +59,15 @@ type DialogContentProps = React.ComponentProps<typeof DialogPrimitive["Content"]
...
-    </DialogPrimitive.Content>
+    <DialogPrimitive.Portal>
+      <DialogPrimitive.Overlay className="fixed inset-0 z-40 bg-gray-500 bg-opacity-75 transition-opacity" />
+      <DialogPrimitive.Content
+        {...props}
+        className="fixed left-1/2 top-1/2 z-50 min-w-[360px] -translate-x-1/2 -translate-y-1/2 rounded bg-white p-6 text-left shadow-xl sm:w-full sm:max-w-lg sm:align-middle"
+        ref={forwardedRef}>
+        {children}
+      </DialogPrimitive.Content>
+    </DialogPrimitive.Portal>
```
Comment (leog) [inline apps/web/components/Dialog.tsx:70]: Same change coming from #2081 which I merged into the branch. The change here to add the `Dialog.Portal` is to be able to attach the dialog to the `body` and not locally where the tag is used, that gets the dialog to be full page and centered which was the desired behavior.

File: apps/web/components/Shell.tsx [hunk 2]
```diff
@@ -31,12 +37,6 @@ import { trpc } from "@lib/trpc";
 import CustomBranding from "@components/CustomBranding";
 import Loader from "@components/Loader";
 import { HeadSeo } from "@components/seo/head-seo";
-import Dropdown, {
-  DropdownMenuContent,
-  DropdownMenuItem,
-  DropdownMenuSeparator,
-  DropdownMenuTrigger,
-} from "@components/ui/Dropdown";
```
Comment (leog) [inline apps/web/components/Shell.tsx:39]: Some reordering of the imports, probably the linter.

File: apps/web/components/eventtype/CreateEventType.tsx [hunk 3]
```diff
@@ -56,12 +56,23 @@ export default function CreateEventTypeButton(props: Props) {
...
   const pageSlug = router.query.eventPage || props.options[0].slug;
   const hasTeams = !!props.options.find((option) => option.teamId);
+  const title: string =
+    typeof router.query.title === "string" && router.query.title ? router.query.title : "";
+  const length: number =
+    typeof router.query.length === "string" && router.query.length ? parseInt(router.query.length) : 15;
+  const description: string =
+    typeof router.query.description === "string" && router.query.description ? router.query.description : "";
+  const slug: string = typeof router.query.slug === "string" && router.query.slug ? router.query.slug : "";
+  const type: string = typeof router.query.type == "string" && router.query.type ? router.query.type : "";
```
Comment (leog) [inline apps/web/components/eventtype/CreateEventType.tsx:67]: Here comes the changes explained in #2081 PR.

File: apps/web/components/ui/TableActions.tsx [hunk 4]
```diff
@@ -38,7 +39,36 @@ const TableActions: FC<Props> = ({ actions }) => {
...
+                  {action.label}
+                </Button>
+              </DropdownMenuItem>
+            ))}
+          </DropdownMenuContent>
+        </Dropdown>
+      </div>
+    </>
+  );
+};
```
Comment (leog) [inline apps/web/components/ui/TableActions.tsx:66]: Transforming the old headlessui `Menu` to the existing `Dropdown` menu.

File: apps/web/package.json [hunk 5]
```diff
@@ -34,7 +34,6 @@
     "@calcom/tsconfig": "*",
     "@calcom/ui": "*",
     "@daily-co/daily-js": "^0.21.0",
-    "@headlessui/react": "^1.4.2",
```
Comment (leog) [inline apps/web/package.json:37]: Bye bye headlessui!

File: apps/web/pages/event-types/index.tsx [hunk 6]
```diff
@@ -183,13 +240,71 @@ const EventTypeList = ({ readOnly, types, profile }: EventTypeListProps): JSX.El
...
+                              onConfirm={(e) => {
+                                e.preventDefault();
+                                deleteEventTypeHandler(type.id);
+                              }}>
+                              {t("delete_event_type_description")}
+                            </ConfirmationDialogContent>
+                          </Dialog>
+                        </DropdownMenuItem>
+                      </DropdownMenuContent>
+                    </Dropdown>
```
Comment (leog) [inline apps/web/pages/event-types/index.tsx:307]: Changes from the other PR 😓

File: apps/web/tailwind.config.js [hunk 7]
```diff
@@ -1 +1,5 @@
-module.exports = require("@calcom/config/tailwind-preset");
+const base = require("@calcom/config/tailwind-preset");
+module.exports = {
+  ...base,
+  content: [...base.content, "../../packages/ui/**/*.{js,ts,jsx,tsx}"],
+};
```
Comment (leog) [inline apps/web/tailwind.config.js:5]: This is the way I found to customize the `tailwing` config from the monorepo without having to bring the contents and plugins entirely to here as before.

File: packages/ui/Button.tsx [hunk 8]
```diff
@@ -52,9 +52,9 @@ export const Button = forwardRef<HTMLAnchorElement | HTMLButtonElement, ButtonPr
         // base styles independent what type of button it is
         "inline-flex items-center",
         // different styles depending on size
-        size === "sm" && "px-3 py-2 text-sm leading-4 font-medium rounded-sm",
-        size === "base" && "px-3 py-2 text-sm font-medium rounded-sm",
-        size === "lg" && "px-4 py-2 text-base font-medium rounded-sm",
```
Comment (leog) [inline packages/ui/Button.tsx:57]: Adjusting the UI to be more consistent to the figma

File: packages/ui/Dropdown.tsx [hunk 9]
```diff
@@ -12,21 +12,24 @@ export const DropdownMenuTrigger = forwardRef<HTMLButtonElement, DropdownMenuTri
       className={
         props.asChild
           ? className
-          : `relative inline-flex items-center rounded-sm bg-transparent px-3 py-2 text-sm font-medium text-gray-700 hover:bg-gray-50 focus:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-neutral-500 focus:ring-offset-1 group-hover:text-black ltr:ml-2 rtl:mr-2 ${className}`
+          : `inline-flex items-center rounded-sm bg-transparent px-3 py-2 text-sm font-medium text-gray-700 hover:bg-gray-50 focus:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-neutral-500 focus:ring-offset-1 group-hover:text-black ${className}`
```
Comment (leog) [inline packages/ui/Dropdown.tsx:15]: This was mesing around with the zindex of the menu trigger now that the dropdowns are by default added under `body` thanks to the Portal.

**Comment by leog** [comment]
@zomars @PeerRich done! and as I reverted creating smaller components when not needed in the #2081 PR and this branch is based on that branch, I also brought those changes here so that's why there are quite a few changes not belonging in this PR, but once merged, there shouldn't be any harm.

File: apps/web/components/eventtype/CreateEventType.tsx (hunk 3 above)
Comment (PeerRich) [reply apps/web/components/eventtype/CreateEventType.tsx:67]: does that mean we can close #2081?

File: apps/web/package.json (hunk 5 above)
Comment (PeerRich) [reply apps/web/package.json:37]: nice!

**Review by PeerRich — APPROVED** [review]

File: apps/web/components/eventtype/CreateEventType.tsx (hunk 3 above)
Comment (leog) [reply apps/web/components/eventtype/CreateEventType.tsx:67]: I guess we can skip that other PR, yes, sorry for the mess!

**Comment by leog** [comment]
Closing to recreate, changes merged from #2081 are causing problems.

**Pull request closed without merging**